File: app/src/main/java/com/qingmei2/notificationdemo/SecondActivity.java
package com.qingmei2.notificationdemo;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        String name = getIntent().getStringExtra("name");

        TextView tv = findViewById(R.id.tv);
        tv.append("\n" + "name = " + name);
    }
}

File: app/src/main/java/com/qingmei2/notificationdemo/SecondActivity.kt
package com.qingmei2.notificationdemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_second.*

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val name = intent.getStringExtra("name")
        tv.append("\nname = $name")
    }
}
